File: chap1/SecondDegre.py
# Ces programmes sont sous licence CeCILL-B V1.
from math import sqrt

a = float(input())
b = float(input())
c = float(input())
delta = b * b - 4 * a * c
if delta < 0.0:
   print("Pas de solution")
elif delta == 0.0:
   # Dangereux (essayer 0.9 0.6 0.1)
   print("Une solution :",- b / (2 * a))
else:
   print("Deux solutions : ",end="")
   print((- b - sqrt(delta)) / (2 * a),end="")
   print(" et ",end="")
   print((- b + sqrt(delta)) / (2 * a))

File: chap1/SecondDegre.java
// Ces programmes sont sous licence CeCILL-B V1.
class SecondDegre {

  public static void main (String [] args) {
    double a,b,c,delta;
    a = Isn.readDouble();
    b = Isn.readDouble();
    c = Isn.readDouble();
    delta = b * b - 4 * a * c;
    if (delta < 0.0) {
      System.out.println("Pas de solution");}
    else {
      if (delta == 0.0) {
        System.out.print("Une solution : ");
        System.out.println(- b / (2 * a));}
      else {
        System.out.print("Deux solutions : ");
        System.out.print((- b - Math.sqrt(delta)) / (2 * a));
        System.out.print(" et ");
        System.out.println((- b + Math.sqrt(delta)) / (2 * a));}}}}
